test/Plutip/Staking.purs

```diff
@@ -0,0 +1,671 @@
+              lookups = mempty
+
+            ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+            liftedE (balanceTx ubTx) >>= signTransaction >>= submitAndLog
+
+          -- List pools
+          poolId <- do
+            pools <- getPoolIds
+            logInfo' "Pool IDs:"
+            logInfo' $ show pools
+            for_ pools \poolId -> do
+              logInfo' "Pool parameters"
+              logInfo' <<< show =<< getPoolParameters poolId
+            liftM (error "unable to get any pools") (pools Array.!! 2)
+
+          -- Delegate
+          do
+            let
+              constraints =
+                mustDelegateStakePlutusScript validator unitRedeemer poolId
+
+              lookups :: Lookups.ScriptLookups Void
+              lookups =
+                Lookups.ownPaymentPubKeyHash alicePkh <>
+                  Lookups.ownStakePubKeyHash aliceStakePkh
+            ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+            liftedE (balanceTx ubTx) >>= signTransaction >>= submitAndLog
+
+          -- Wait until rewards
+          let
+            -- No need for limit on number of retries, because we have a
+            -- timeout for tests.
+            waitUntilRewards = do
+              mbDelegationsAndRewards <-
+                getValidatorHashDelegationsAndRewards stakeValidatorHash
+              case mbDelegationsAndRewards of
+                Just dels@{ rewards } | unwrap <$> rewards > Just zero ->
+                  pure dels
+                _ -> do
+                  liftAff $ delay $ Milliseconds 5000.0
+                  waitUntilRewards
+
+          { rewards: rewardsBefore, delegate } <- waitUntilRewards
+          delegate `shouldEqual` Just poolId
+
+          -- Withdraw
+          do
+            let
+              constraints =
+                mustWithdrawStakePlutusScript validator unitRedeemer
+
+              lookups :: Lookups.ScriptLookups Void
+              lookups =
+                Lookups.ownPaymentPubKeyHash alicePkh <>
+                  Lookups.ownStakePubKeyHash aliceStakePkh
+            ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+            liftedE (balanceTx ubTx) >>= signTransaction >>= submitAndLog
+
+          -- Check rewards.
+          -- Not going to deregister here, because the rewards are added too
+          -- soon, and we can't deregister the stake key if there are rewards
+          -- left.
+          -- This will not happen in real life scenarios, because epoch are
+          -- (usually) significantly longer.
+          do
+            { rewards: rewardsAfter } <- liftedM "Unable to get rewards" $
+              getValidatorHashDelegationsAndRewards stakeValidatorHash
+            rewardsAfter `shouldSatisfy` \after -> after < rewardsBefore
+
+    test "Native Stake script: delegate to existing pool & withdraw rewards" do
+      let
+        distribution =
+          withStakeKey privateStakeKey
```

Comment:
```suggestion
```

If she has the same stake key as bob, it's no longer just bob who can withdraw against the validator?